src/main/java/com/intel/gkl/compression/IntelInflater.java

```diff
@@ -141,7 +152,16 @@ public void setInput(byte[] b, int off, int len) throws NullPointerException {
      *         output buffer
      */
     @Override
-    public int inflate (byte[] b, int off, int len ) {
+    public int inflate (byte[] b, int off, int len ) throws NullPointerException {
+        if(b == null) {
+            throw new NullPointerException("Input is null");
+        }
+        if(off < 0 || off > b.length - len) {
+            throw new NullPointerException("Input offset is not valid.");
+        }
+        if(len <= 0) {
+            throw new NullPointerException("Input buffer length is zero.");
```

Comment:
same

Reply:
Done .

Reply:
Is there a new patch? I still see 

if(len <= 0) {
            throw new NullPointerException("Input buffer length is zero.");